allensdk/brain_observatory/gaze_mapping/_eye_calibration.py

```diff
@@ -0,0 +1,336 @@
+            self.monitor_position)
+        R_monitor = object_rotation_matrix(self.monitor_rotations[0],
+                                           self.monitor_rotations[1],
+                                           self.monitor_rotations[2])
+
+        result = np.dot(R_monitor.T,
+                        np.dot(R_monitor_to_eye.T, monitor_positions.T))
+        return result[:2].T
+
+    def pupil_position_on_monitor_in_degrees(self, pupil_parameters,
+                                             cr_parameters):
+        '''Get pupil position on monitor measured in visual degrees.
+
+        Parameters
+        ----------
+        pupil_parameters :  numpy.ndarray
+            Array of pupil parameters for each eye tracking frame.
+        cr_paramaeters : numpy.ndarray
+            Array of corneal reflection parameters for each eye
+            tracking frame.
+
+        Returns
+        -------
+        numpy.ndarray
+            Pupil position estimate in visual degrees.
+        '''
+
+        mag = np.sqrt(np.sum(self.monitor_position**2))
+
+        pupil_pos = self.pupil_position_on_monitor_in_cm(pupil_parameters,
+                                                         cr_parameters)
+
+        x = pupil_pos.T[0]
+        y = pupil_pos.T[1]
+
+        meridian = np.arctan(x/mag)*180/np.pi
+        elevation = np.arctan(y/np.sqrt(mag**2 + x**2))*180/np.pi
+
+        angles = np.vstack([meridian, elevation]).T
+
+        return angles
+
+
+def compute_ellipse_areas(ellipse_params) -> pd.Series:
+    """Compute circular area of a pupil or eye ellipse using major axis.
+
+    Assume the pupil/eye is a circle, and that as it moves off-axis
+    with the camera, the observed ellipse major axis remains the
+    diameter of the circle.
+
+    Args:
+        ellipse_params (pandas.DataFrame): A table of pupil/eye parameters
+            consisting of 5 columns ("center_x", "center_y", "height",
+            "phi", "width") and n-row timepoints.
+
+    Returns:
+        pandas.Series: A series of pupil/eye areas for n-timepoints.
+    """
+
+    # Take the biggest value between height and width columns and
+    # assume that it is the pupil/eye circle diameter.
+    diameter = ellipse_params[["height", "width"]].max(axis=1)
+    radii = diameter / 2
+    return np.pi * radii * radii
+
+
+def project_to_plane(plane_normal, plane_point, points):
+    '''Project from the origin through points onto a plane.
+
+    Parameters
+    ----------
+    plane_normal : numpy.ndarray
+        [x, y, z] normal unit vector to the plane.
+    plane_point : numpy.ndarray
+        [x, y, z] point on the plane.
+    points : numpy.ndarray
+        [nx3] points in space through which to project.
+
+    Returns
+    -------
+    numpy.ndarray
+        [nx3] points projected on the plane.
+    '''
+    factor = (np.sum(plane_normal * plane_point) /
+              np.sum(plane_normal * points, axis=1))
+    return (factor * points.T).T
```

Comment:
I am pretty sure that this is not correct. This formula gives a length along the line towards the plane not a projection onto a different basis. A good indicator that this is wrong is that all points are used to find a `factor` and then multiplied to find a transformation. Why would one need to use all points in order to find a projection? We should be able to project each point separately, but not with this expression.

Reply:
I think it's doing the right thing. factor ends up being an array that is the same length number of points [n x 3(x, y, z)].

Equivalent to:
```
factor = (np.sum(plane_normal * plane_point) /
              np.sum(plane_normal * points, axis=1))

factors = np.dot(plane_point, plane_normal) / np.dot(points, plane_normal)

assert np.allclose(factor, factors, equal_nan=True)
```